omnigibson/prims/prim_base.py

```diff
@@ -20,43 +21,44 @@ class BasePrim(Serializable, UniquelyNamed, Recreatable, ABC):
         unless it is a non-root articulation link.
 
     Args:
-        prim_path (str): prim path of the Prim to encapsulate or create.
+        relative_prim_path (str): prim path of the Prim to encapsulate or create.
         name (str): Name for the object. Names need to be unique per scene.
         load_config (None or dict): If specified, should contain keyword-mapped values that are relevant for
             loading this prim at runtime. Note that this is only needed if the prim does not already exist at
-            @prim_path -- it will be ignored if it already exists. Subclasses should define the exact keys expected
+            @relative_prim_path -- it will be ignored if it already exists. Subclasses should define the exact keys expected
             for their class.
     """
 
     def __init__(
         self,
-        prim_path,
+        relative_prim_path,
         name,
         load_config=None,
     ):
-        self._prim_path = prim_path
+        self._relative_prim_path = relative_prim_path
         self._name = name
         self._load_config = dict() if load_config is None else load_config
 
         # Other values that will be filled in at runtime
+        self._scene = None
```

Comment:
Not sure if this is a bad thing, but this currently means there is a circular reference between both the scene and the object. Maybe this is ok? What do you all think @ChengshuLi @cgokmen  @hang-yin ?

Reply:
I think it's fine tbh. I know Roberto also had strong feelings about this but I don't fully understand the issue with circular references, this is not a reference counting language after all

Reply:
TODO: maybe keep a scene index reference instead of the scene object?